core/src/test/java/org/apache/iceberg/TestOffsetsBasedSplitScanTaskIterator.java

```diff
@@ -62,15 +62,15 @@ private static void verify(
             offsetRanges,
             TestOffsetsBasedSplitScanTaskIterator::createSplitTask);
     List<FileScanTask> tasks = Lists.newArrayList(splitTaskIterator);
-    Assert.assertEquals("Number of tasks don't match", offsetLenPairs.size(), tasks.size());
+    Assertions.assertThat(tasks).as("Number of tasks don't match").hasSize(offsetLenPairs.size());
```

Comment:
maybe use `hasSameSizeAs()` here?

Reply:
I've modified the code based on your suggestions ` hasSameSizeAs()` .